File: Python/ReverseDegreeofaString.py
class Solution:
    def reverseDegree(self, s: str) -> int:

        alphabets = "abcdefghijklmnopqrstuvwxyz"
        reversed = {}

        for i in range(len(alphabets)):
            reversed[alphabets[i]] = len(alphabets) - i

        output = 0

        for i in range(len(s)):

            product = reversed[s[i]] * (i + 1)
            output += product

        return output


solution = Solution()

print(solution.reverseDegree("abc"))

File: JavaScript/ReverseDegreeofaString.js
const reverseDegree = function (s) {
  const alphabets = "abcdefghijklmnopqrstuvwxyz";

  const reversed = {};

  const n = alphabets.length;

  for (let i = 0; i < n; i++) {
    reversed[alphabets[i]] = n - i;
  }

  let output = 0;

  for (let i = 0; i < s.length; i++) {
    const ch = s[i];
    const index = i + 1;
    const reversedIndex = reversed[ch];
    const product = reversedIndex * index;
    output += product;
  }

  return output;
};

console.log(reverseDegree("abc"));
